<table align="center" width=100%>
    <tr>
        </td>
            </td>
        </td>
        <td>
            <div align="center">
                <font color="#21618C" size=8px>
                    <b> Regression modeling on House Price Data
                    </b>
                </font>
            </div>
        </td>
    </tr>
</table>

# Problem Statement

This dataset is created for the prediction of House Prices from an Indian perspective Towns. The dataset contains various features that are important during the application for the selling the Houses. The predicted output obtained from the classification algorithm gives a fair idea about the chances of a Houses Sold.

# Data Dictionary 
 
 The Data set contains 506 observations of House Price From Different Towns. Corresponding to each house price,data of other veriables is available on which price is suspected to depend.

## About the dataset (House Pricing prediction data) 

**price:** Value of the house 

**crime_rate:** Crime rate in that neighborhood

**resid_area:** Preporsion of residential area in the town

**air_qual:** Quality of Air in that neighborhood

**room_num:** Avearge number of rooms in houses of that locality

**age:** How old is the house counstuction  in years

**dist1:** Distance from employment hub 1

**dist2:** Distance from employment hub 2

**dist3:** Distance from employment hub 3

**dist4:** Distance from employment hub 4

**teachers:** Number of Teachers per 1000 population in the town

**poor_prop:** Preportion of poor population in the town

**airport:** Is there an airport in city? (Yes/No)

**n_hos_beds:** Number of Hospital Beds per 1000 population in the town

**n_hot_rooms:** Number of Hotel rooms per 1000 population in town

**waterbody:** What type of natural fresh water sourcs is there in the city (Lake/River/Both/None)

**rainfall:** The yearly average rainfall in centimeters

**bus_ter:** Is there a bus terminal in city? (Yes/No)

**parks:** Proportion of land assigned as park and green areas in town

In [1]:
# import 'Pandas' 
import pandas as pd 

# import 'Numpy' 
import numpy as np

# import subpackage of Matplotlib
import matplotlib.pyplot as plt
from matplotlib.colors import ListedColormap

# import 'Seaborn' 
import seaborn as sns

# to suppress warnings 
from warnings import filterwarnings
filterwarnings('ignore')

# display all columns of the dataframe
pd.options.display.max_columns = None

# display all rows of the dataframe
pd.options.display.max_rows = None

# set the plot size using 'rcParams'
# once the plot size is set using 'rcParams', it sets the size of all the forthcoming plots in the file
# pass width and height in inches to 'figure.figsize' 
plt.rcParams['figure.figsize'] = [15,8]

In [2]:
# import train-test split 
from sklearn.model_selection import train_test_split

In [3]:
# load the csv file
# store the data in 'df_houseprice'
df_houseprice = pd.read_csv("House_Price.csv")

# display first five observations using head()
df_houseprice.head()

price  crime_rate  resid_area  air_qual  room_num   age  dist1  dist2  \
0   24.0     0.00632       32.31     0.538     6.575  65.2   4.35   3.81   
1   21.6     0.02731       37.07     0.469     6.421  78.9   4.99   4.70   
2   34.7     0.02729       37.07     0.469     7.185  61.1   5.03   4.86   
3   33.4     0.03237       32.18     0.458     6.998  45.8   6.21   5.93   
4   36.2     0.06905       32.18     0.458     7.147  54.2   6.16   5.86   

   dist3  dist4  teachers  poor_prop airport  n_hos_beds  n_hot_rooms  \
0   4.18   4.01      24.7       4.98     YES       5.480      11.1920   
1   5.12   5.06      22.2       9.14      NO       7.332      12.1728   
2   5.01   4.97      22.2       4.03      NO       7.394     101.1200   
3   6.16   5.96      21.3       2.94     YES       9.268      11.2672   
4   6.37   5.86      21.3       5.33      NO       8.824      11.2896   

  waterbody  rainfall bus_ter     parks  
0     River        23     YES  0.049347  
1      Lake        42     YES  0.046146  
2       NaN        38     YES  0.045764  
3      Lake        45     YES  0.047151  
4      Lake        55     YES  0.039474

In [4]:
df_houseprice.bus_ter.value_counts()

bus_ter
YES    506
Name: count, dtype: int64

In [5]:
# So as we see here Bus_ter column have only one unique value that means it wont usefull for our analysis
# Axis  = 1  use for column 0 for row
# inplace = True indecate permanat changes
df_houseprice.drop("bus_ter",axis =1,inplace = True)

In [6]:
# sort the variables on the basis of total null values in the variable
# 'isnull().sum()' returns the number of missing values in each variable
# 'ascending = False' sorts values in the descending order
# the variable with highest number of missing values will appear first
Total = df_houseprice.isnull().sum().sort_values(ascending=False)          

# calculate percentage of missing values
# 'ascending = False' sorts values in the descending order
# the variable with highest percentage of missing values will appear first
Percent = (df_houseprice.isnull().sum()*100/df_houseprice.isnull().count()).sort_values(ascending=False)   

# concat the 'Total' and 'Percent' columns using 'concat' function
# pass a list of column names in parameter 'keys' 
# 'axis = 1' concats along the columns
missing_data = pd.concat([Total, Percent], axis = 1, keys = ['Total', 'Percentage of Missing Values'])    
missing_data

Total  Percentage of Missing Values
waterbody      155                     30.632411
n_hos_beds       8                      1.581028
price            0                      0.000000
crime_rate       0                      0.000000
rainfall         0                      0.000000
n_hot_rooms      0                      0.000000
airport          0                      0.000000
poor_prop        0                      0.000000
teachers         0                      0.000000
dist4            0                      0.000000
dist3            0                      0.000000
dist2            0                      0.000000
dist1            0                      0.000000
age              0                      0.000000
room_num         0                      0.000000
air_qual         0                      0.000000
resid_area       0                      0.000000
parks            0                      0.000000

In [7]:
df_houseprice.fillna(df_houseprice["n_hos_beds"].mean(),inplace=True)

In [8]:
# sort the variables on the basis of total null values in the variable
# 'isnull().sum()' returns the number of missing values in each variable
# 'ascending = False' sorts values in the descending order
# the variable with highest number of missing values will appear first
Total = df_houseprice.isnull().sum().sort_values(ascending=False)          

# calculate percentage of missing values
# 'ascending = False' sorts values in the descending order
# the variable with highest percentage of missing values will appear first
Percent = (df_houseprice.isnull().sum()*100/df_houseprice.isnull().count()).sort_values(ascending=False)   

# concat the 'Total' and 'Percent' columns using 'concat' function
# pass a list of column names in parameter 'keys' 
# 'axis = 1' concats along the columns
missing_data = pd.concat([Total, Percent], axis = 1, keys = ['Total', 'Percentage of Missing Values'])    
missing_data

Total  Percentage of Missing Values
price            0                           0.0
crime_rate       0                           0.0
rainfall         0                           0.0
waterbody        0                           0.0
n_hot_rooms      0                           0.0
n_hos_beds       0                           0.0
airport          0                           0.0
poor_prop        0                           0.0
teachers         0                           0.0
dist4            0                           0.0
dist3            0                           0.0
dist2            0                           0.0
dist1            0                           0.0
age              0                           0.0
room_num         0                           0.0
air_qual         0                           0.0
resid_area       0                           0.0
parks            0                           0.0

In [9]:
# store the target variable 'Chance of Admit' in a dataframe 'df_target'
df_target = df_houseprice['price']

# store all the independent variables in a dataframe 'df_feature'
# drop the column 'Chance of Admit' using drop()
# 'axis = 1' drops the specified column
df_feature = df_houseprice.drop('price', axis = 1)

In [10]:
df_target.head()

0    24.0
1    21.6
2    34.7
3    33.4
4    36.2
Name: price, dtype: float64

In [11]:
df_feature.head()

crime_rate  resid_area  air_qual  room_num   age  dist1  dist2  dist3  \
0     0.00632       32.31     0.538     6.575  65.2   4.35   3.81   4.18   
1     0.02731       37.07     0.469     6.421  78.9   4.99   4.70   5.12   
2     0.02729       37.07     0.469     7.185  61.1   5.03   4.86   5.01   
3     0.03237       32.18     0.458     6.998  45.8   6.21   5.93   6.16   
4     0.06905       32.18     0.458     7.147  54.2   6.16   5.86   6.37   

   dist4  teachers  poor_prop airport  n_hos_beds  n_hot_rooms waterbody  \
0   4.01      24.7       4.98     YES       5.480      11.1920     River   
1   5.06      22.2       9.14      NO       7.332      12.1728      Lake   
2   4.97      22.2       4.03      NO       7.394     101.1200  7.899767   
3   5.96      21.3       2.94     YES       9.268      11.2672      Lake   
4   5.86      21.3       5.33      NO       8.824      11.2896      Lake   

   rainfall     parks  
0        23  0.049347  
1        42  0.046146  
2        38  0.045764  
3        45  0.047151  
4        55  0.039474

In [12]:
# filter the numerical features in the dataset
# 'select_dtypes' is used to select the variables with given data type
# 'include = [np.number]' will include all the numerical variables
df_num = df_feature.select_dtypes(include = [np.number])

# display numerical features
df_num.head()

crime_rate  resid_area  air_qual  room_num   age  dist1  dist2  dist3  \
0     0.00632       32.31     0.538     6.575  65.2   4.35   3.81   4.18   
1     0.02731       37.07     0.469     6.421  78.9   4.99   4.70   5.12   
2     0.02729       37.07     0.469     7.185  61.1   5.03   4.86   5.01   
3     0.03237       32.18     0.458     6.998  45.8   6.21   5.93   6.16   
4     0.06905       32.18     0.458     7.147  54.2   6.16   5.86   6.37   

   dist4  teachers  poor_prop  n_hos_beds  n_hot_rooms  rainfall     parks  
0   4.01      24.7       4.98       5.480      11.1920        23  0.049347  
1   5.06      22.2       9.14       7.332      12.1728        42  0.046146  
2   4.97      22.2       4.03       7.394     101.1200        38  0.045764  
3   5.96      21.3       2.94       9.268      11.2672        45  0.047151  
4   5.86      21.3       5.33       8.824      11.2896        55  0.039474

In [13]:
# filter the categorical features in the dataset
# 'select_dtypes' is used to select the variables with given data type
# 'include = [np.object]' will include all the categorical variables
df_cat = df_feature.select_dtypes(include = ['object'])

# display categorical features
df_cat.columns

Index(['airport', 'waterbody'], dtype='object')

In [14]:
# use 'get_dummies' from pandas to create dummy variables
# use 'drop_first' to create (n-1) dummy variables
dummy_var = pd.get_dummies(data = df_cat, drop_first = True)
dummy_var = dummy_var.reset_index()

In [15]:
# use 'get_dummies' from pandas to create dummy variables
# use 'drop_first' to create (n-1) dummy variables
dummy_var2 = pd.get_dummies(data = df_cat)

In [16]:
dummy_var2.head(2)

airport_NO  airport_YES  waterbody_7.899767068273092  waterbody_Lake  \
0       False         True                        False           False   
1        True        False                        False            True   

   waterbody_Lake and River  waterbody_River  
0                     False             True  
1                     False            False

In [17]:
dummy_var.head(3)

index  airport_YES  waterbody_Lake  waterbody_Lake and River  \
0      0         True           False                     False   
1      1        False            True                     False   
2      2        False           False                     False   

   waterbody_River  
0             True  
1            False  
2            False

In [18]:
# import StandardScaler to perform scaling
from sklearn.preprocessing import StandardScaler 

# initialize the standard scalar
X_scaler = StandardScaler()

# scale all the numerical columns
# standardize all the columns of the dataframe 'df_num'
num_scaled = X_scaler.fit_transform(df_num)

# create a dataframe of scaled numerical variables
# pass the required column names to the parameter 'columns'
df_num_scaled = pd.DataFrame(num_scaled, columns = df_num.columns)
df_num_scaled = df_num_scaled.reset_index()

In [19]:
df_num_scaled.head(3)

index  crime_rate  resid_area  air_qual  room_num       age     dist1  \
0      0   -0.419782   -1.287909 -0.144217  0.413672 -0.120013  0.179451   
1      1   -0.417339   -0.593381 -0.740262  0.194274  0.367166  0.483280   
2      2   -0.417342   -0.593381 -0.740262  1.282714 -0.265812  0.502269   

      dist2     dist3     dist4  teachers  poor_prop  n_hos_beds  n_hot_rooms  \
0  0.086032  0.103569  0.186459  1.459000  -1.075562   -1.653421    -0.353398   
1  0.508534  0.547446  0.687144  0.303094  -0.492439   -0.387954    -0.166000   
2  0.584490  0.495503  0.644228  0.303094  -1.208727   -0.345589    16.828839   

   rainfall     parks  
0 -1.294408 -0.480763  
1  0.225431 -0.782183  
2 -0.094535 -0.818115

In [20]:
# concat the dummy variables with numeric features to create a dataframe of all independent variables
# 'axis=1' concats the dataframes along columns 
X = pd.concat([df_num_scaled, dummy_var], axis = 1)

# display first five observations
X.head()

index  crime_rate  resid_area  air_qual  room_num       age     dist1  \
0      0   -0.419782   -1.287909 -0.144217  0.413672 -0.120013  0.179451   
1      1   -0.417339   -0.593381 -0.740262  0.194274  0.367166  0.483280   
2      2   -0.417342   -0.593381 -0.740262  1.282714 -0.265812  0.502269   
3      3   -0.416750   -1.306878 -0.835284  1.016303 -0.809889  1.062454   
4      4   -0.412482   -1.306878 -0.835284  1.228577 -0.511180  1.038717   

      dist2     dist3     dist4  teachers  poor_prop  n_hos_beds  n_hot_rooms  \
0  0.086032  0.103569  0.186459  1.459000  -1.075562   -1.653421    -0.353398   
1  0.508534  0.547446  0.687144  0.303094  -0.492439   -0.387954    -0.166000   
2  0.584490  0.495503  0.644228  0.303094  -1.208727   -0.345589    16.828839   
3  1.092442  1.038545  1.116302 -0.113032  -1.361517    0.934910    -0.339029   
4  1.059212  1.137709  1.068618 -0.113032  -1.026501    0.631526    -0.334750   

   rainfall     parks  index  airport_YES  waterbody_Lake  \
0 -1.294408 -0.480763      0         True           False   
1  0.225431 -0.782183      1        False            True   
2 -0.094535 -0.818115      2        False           False   
3  0.465405 -0.687571      3         True            True   
4  1.265320 -1.410280      4        False            True   

   waterbody_Lake and River  waterbody_River  
0                     False             True  
1                     False            False  
2                     False            False  
3                     False            False  
4                     False            False

In [21]:
# import various functions from statsmodels
import statsmodels
import statsmodels.api as sm

# add a constant column to the dataframe
# while using the 'Logit' method in the Statsmodels library, the method do not consider the intercept by default
# we can add the intercept to the set of independent variables using 'add_constant()'
X = sm.add_constant(X)

# split data into train subset and test subset
# set 'random_state' to generate the same dataset each time you run the code 
# 'test_size' returns the proportion of data to be included in the testing set
X_train, X_test, y_train, y_test = train_test_split(X, 
                                                    df_target, 
                                                    random_state = 10,
                                                    test_size = 0.2)

# check the dimensions of the train & test subset using 'shape'
# print dimension of train set
print('X_train', X_train.shape)
print('y_train', y_train.shape)

# print dimension of test set
print('X_test', X_test.shape)
print('y_test', y_test.shape)

X_train (404, 22)
y_train (404,)
X_test (102, 22)
y_test (102,)


# Linear Regression


In [22]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Create a linear regression model
model = LinearRegression()

# Fit the model to the training data
model.fit(X_train, y_train)

# Predict on the testing data
predictions = model.predict(X_test)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 32.59841350908501


In [23]:
predictions

array([27.75382029, 30.31458208, 27.59776514, 23.83408658, 19.22606257,
       17.09679089, 37.89799566, 15.99983184, 23.1870346 , 37.81809287,
       22.28249766, 29.72391843, 27.73401777, 31.96563064, 30.18707561,
       41.12004435, 25.02156448, 23.8650445 , 27.46694607, 20.35421787,
       34.27278287, 16.86559178, 26.46876771, 25.88304839, 35.20322674,
       20.97962791, 18.96646146, 16.78127836, 38.19489347,  1.99005852,
       32.52178313, 31.40674083, 26.19044702, 25.31561769, 18.51186123,
       21.21081781,  5.66219887, 30.59215227, 28.50549982, 27.48929402,
       36.23171596, 30.05630961, 15.23555015, 31.20326613, 18.08709741,
       28.87648953, 19.39538312, 19.96727535, 37.20604846, 17.69229475,
       27.10151394, 16.64499059, 24.17844987, 36.59474455, 28.36077558,
       34.18438675, 22.60367786, 18.9895157 , 18.79770291, 23.04535088,
       20.65615579, 23.38182391, 39.25870921, 43.0500349 , 30.97547136,
       18.30478953, 23.52748036,  1.05487737, 32.73900062, 30.26

# Ridge Model

In [24]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import Ridge
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Create a Ridge regression model
ridge_model = Ridge(alpha=1.0)  # alpha is the regularization strength

# Fit the model to the training data
ridge_model.fit(X_train, y_train)

# Predict on the testing data
predictions = ridge_model.predict(X_test)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 32.00974705332775


# Lasso Model

In [25]:
from sklearn.model_selection import train_test_split
from sklearn.linear_model import Lasso
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Create a Lasso regression model
lasso_model = Lasso(alpha=1.0)  # alpha is the regularization strength

# Fit the model to the training data
lasso_model.fit(X_train, y_train)

# Predict on the testing data
predictions = lasso_model.predict(X_test)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 40.47972772114124


# Decision Tree Regressor

In [26]:
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeRegressor
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Create a Decision Tree regression model
tree_model = DecisionTreeRegressor()

# Fit the model to the training data
tree_model.fit(X_train, y_train)

# Predict on the testing data
predictions = tree_model.predict(X_test)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 27.854411764705883


# Random Forest Regressor

In [27]:
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Create a Random Forest regression model
forest_model = RandomForestRegressor(n_estimators=100, random_state=42)

# Fit the model to the training data
forest_model.fit(X_train, y_train)

# Predict on the testing data
predictions = forest_model.predict(X_test)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 14.684267696078422


# Gaussian Process Regressor

In [28]:
from sklearn.gaussian_process import GaussianProcessRegressor
from sklearn.gaussian_process.kernels import RBF
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Create a Gaussian Process regression model with an RBF kernel
kernel = 1.0 * RBF(length_scale=1.0)
gp_model = GaussianProcessRegressor(kernel=kernel, n_restarts_optimizer=10, random_state=42)

# Fit the model to the training data
gp_model.fit(X_train, y_train)

# Predict on the testing data
predictions, std_dev = gp_model.predict(X_test, return_std=True)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 14.232336119247645


# Polynomial Regression 

In [29]:
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import PolynomialFeatures
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Generate polynomial features
degree = 3  # degree of the polynomial
poly = PolynomialFeatures(degree=degree)
X_train_poly = poly.fit_transform(X_train)
X_test_poly = poly.transform(X_test)

# Create a Polynomial Regression model
poly_model = LinearRegression()

# Fit the model to the training data
poly_model.fit(X_train_poly, y_train)

# Predict on the testing data
predictions = poly_model.predict(X_test_poly)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 44.41430141597724


# Neural Network Regression  


In [30]:
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense
from tensorflow.keras.optimizers import Adam
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Normalize the features
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Create a neural network model
model = Sequential([
    Dense(64, activation='relu', input_shape=(X_train_scaled.shape[1],)),
    Dense(32, activation='relu'),
    Dense(1)
])

# Compile the model
model.compile(optimizer=Adam(), loss='mean_squared_error')

# Fit the model to the training data
history = model.fit(X_train_scaled, y_train, epochs=100, batch_size=32, validation_split=0.2)

# Predict on the testing data
predictions = model.predict(X_test_scaled)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Epoch 1/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 2s 32ms/step - loss: 507.2545 - val_loss: 527.5601
Epoch 2/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - loss: 470.8257 - val_loss: 479.2533
Epoch 3/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 8ms/step - loss: 428.9349 - val_loss: 420.0580
Epoch 4/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 13ms/step - loss: 397.8919 - val_loss: 348.7835
Epoch 5/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 12ms/step - loss: 321.1705 - val_loss: 272.5100
Epoch 6/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 9ms/step - loss: 248.1997 - val_loss: 197.6421
Epoch 7/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 14ms/step - loss: 168.7777 - val_loss: 137.9076
Epoch 8/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 7ms/step - loss: 112.0864 - val_loss: 106.4462
Epoch 9/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - loss: 97.8992 - val_loss: 90.0484
Epoch 10/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 6ms/step - loss: 72.0248 - val_loss: 76.3523
Epoch 11/100
11/11 ━━━━━━━━━━━━━━━━━━━━ 0s 9ms/step - loss: 59.1011 - val_loss: 64.5182
Epoch 12/100
11/11 ━━

# KNN

In [31]:
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsRegressor
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Create a KNN regression model
k = 5  # number of neighbors
knn_model = KNeighborsRegressor(n_neighbors=k)

# Fit the model to the training data
knn_model.fit(X_train, y_train)

# Predict on the testing data
predictions = knn_model.predict(X_test)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 39.810121568627444


# SVR

In [32]:
from sklearn.model_selection import train_test_split
from sklearn.svm import SVR
from sklearn.metrics import mean_squared_error

# Assuming X_train, X_test, y_train, y_test are already defined from train_test_split

# Create a SVM regression model
svm_model = SVR(kernel='rbf')  # 'rbf' kernel is commonly used for non-linear regression

# Fit the model to the training data
svm_model.fit(X_train, y_train)

# Predict on the testing data
predictions = svm_model.predict(X_test)

# Calculate mean squared error
mse = mean_squared_error(y_test, predictions)
print("Mean Squared Error:", mse)


Mean Squared Error: 99.45865077390333


**Guassian Process Regressor Model is found out to be Best Model with this Dataset With MSE of 14.23**
